typings/utils.d.ts

```diff
@@ -0,0 +1,8 @@
+// Functionally the same as Exclude, but for strings only.
+type Diff<T extends string, U extends string> = ({ [P in T]: P } &
```

Comment:
I'm not quite sure what this `Diff` is doing, but I think you'd be better off with:
```ts
export type ExcludeField<T, F extends keyof T> = Pick<T, Exclude<keyof T, F>>;
```
where I presume `Diff` is doing the equivalent of `Exclude<keyof T, F>`?

Reply:
The SDK is still using TS 2.6.2 unfortunately, where `Exclude` isn't available, so I had to use a backwards compatible implementation that is based on mapped types.
Bumping TS w/o a major version can cause CI tests to fails on consumer projects that haven't caught up yet with newer TS versions.
On the other hand the `v12-dev` branch (the next major) is already updated to use `Exclude` as you suggest.
See: https://github.com/balena-io/balena-sdk/blob/v12-dev/typings/utils.d.ts

PS: `Diff` was the most popular community typing helper afaict for doing this, so I also used the same name for reference.